packages/apex-node/test/utils/traceFlags.test.ts

```diff
@@ -0,0 +1,202 @@
+      done: true,
+      totalSize: 0,
+      records: []
+    });
+
+    toolingCreateStub
+      .onFirstCall()
+      .resolves({ success: true, id: 'aBcDeF' })
+      .onSecondCall()
+      .resolves({ success: true, id: '01020304' });
+
+    const ensure = await flags.ensureTraceFlags();
+
+    expect(ensure).to.equal(true);
+    expect(queryStub.callCount, 'Query stub called').to.equal(1);
+    expect(toolingQueryStub.callCount, 'Tooling stub called').to.equal(1);
+
+    let createArgs = toolingCreateStub.getCall(0).args;
+    expect(createArgs[0]).to.equal('DebugLevel');
+    expect(createArgs[1].developerName).to.contain('SFDC_DevConsole');
+    expect(createArgs[1].MasterLabel).to.contain('SFDC_DevConsole');
+    expect(createArgs[1].ApexCode).to.equal('FINEST');
+    expect(createArgs[1].Visualforce).to.equal('FINER');
+
+    createArgs = toolingCreateStub.getCall(1).args;
+    expect(createArgs[0]).to.equal('TraceFlag');
+    expect(createArgs[1].tracedentityid).to.equal(USER_ID);
+    expect(createArgs[1].logtype).to.equal('DEVELOPER_LOG');
+    expect(createArgs[1].debuglevelid).to.equal('aBcDeF');
+    expect(createArgs[1].StartDate).to.equal('');
+    const expDate = new Date(createArgs[1].ExpirationDate);
+    expect(expDate.getTime() - currDate).to.be.greaterThan(60000 * 29);
+  });
+
+  it('should raise error for missing username', async () => {
+    flags = new TraceFlags(mockConnection);
+    sb.stub(mockConnection, 'getUsername').returns(undefined);
+
+    try {
+      await flags.ensureTraceFlags();
+      fail('Expected an error');
+    } catch (err) {
+      expect(err.message).to.equal(nls.localize('error_no_default_username'));
+    }
+  });
+
+  it('should raise error for unknown user', async () => {
+    flags = new TraceFlags(mockConnection);
+    sb.stub(mockConnection, 'query')
+      .onFirstCall()
+      .resolves({ done: true, totalSize: 0, records: [] });
+
+    try {
+      await flags.ensureTraceFlags();
+      fail('Expected an error');
+    } catch (err) {
+      expect(err.message).to.equal(nls.localize('trace_flags_unknown_user'));
+    }
+  });
+
+  it('should raise error on failure to create debug level', async () => {
+    flags = new TraceFlags(mockConnection);
+    queryStub = sb.stub(mockConnection, 'query');
+    toolingCreateStub = sb.stub(mockConnection.tooling, 'create');
+    toolingQueryStub = sb.stub(mockConnection.tooling, 'query');
+
+    queryStub
+      .onFirstCall()
+      .resolves({ done: true, totalSize: 1, records: [{ Id: USER_ID }] });
+    toolingQueryStub.onFirstCall().resolves({
+      done: true,
+      totalSize: 0,
+      records: []
+    });
+    toolingCreateStub.onFirstCall().resolves({ success: false, id: undefined });
+
+    try {
+      await flags.ensureTraceFlags();
+      fail('Expected to raise an error');
+    } catch (err) {
+      expect(err.message).to.equal(
+        nls.localize('trace_flags_failed_to_create_debug_level')
+      );
+    }
+  });
+});
```

Comment:
let's add a test case for how an unexpected error would be handled

Reply:
Sounds good @AnanyaJha! Is there a specific query you think we should test, or an example error elsewhere I can look at? We do have a test for the creation of the debug level failing with an unexpected error.